packages/bsky/src/api/com/atproto/admin/searchRepos.ts

```diff
@@ -16,9 +17,17 @@ export default function (server: Server, ctx: AppContext) {
         throw new InvalidRequestError('The invitedBy parameter is unsupported')
       }
 
+      let searchField: 'did' | 'handle'
+      try {
+        ensureValidDid(term)
+        searchField = 'did'
+      } catch (err) {
+        searchField = 'handle'
+      }
```

Comment:
Throwing and then immediately catching is a bit of an anti-pattern IMO(hidden control flow, error creation overhead etc.), but there's no `isValidDid` method so I guess it's ok

Reply:
Generally we handle this case with:

`switchField = term.startWith('did:') ? 'did' : 'handle'`

Reply:
Felt safer to use `ensureDid` but sure, I can replace this with the `startWith()` approach.